src/snapshot-editor/src/edit_vmstate.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use clap::Subcommand;
+use clap_num::maybe_hex;
+use vmm::arch::aarch64::regs::Aarch64RegisterVec;
+use vmm::persist::MicrovmState;
+
+use crate::utils::{open_vmstate, save_vmstate, UtilsError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum EditVmStateError {
+    #[error("{0}")]
+    Utils(#[from] UtilsError),
+}
+
+#[derive(Debug, Subcommand)]
+pub enum EditVmStateSubCommand {
+    /// Remove registers from vcpu states.
+    #[cfg(target_arch = "aarch64")]
+    RemoveRegs {
+        /// Set of regs to remove.
+        #[arg(value_parser=maybe_hex::<u64>, num_args = 1.., value_delimiter = ' ')]
+        regs: Vec<u64>,
+        /// Path to the vmstate file.
+        #[arg(short, long)]
+        vmstate_path: PathBuf,
+        /// Path of output file.
+        #[arg(short, long)]
+        output_path: PathBuf,
+    },
+}
+
+pub fn edit_vmstate_command(command: EditVmStateSubCommand) -> Result<(), EditVmStateError> {
+    match command {
+        #[cfg(target_arch = "aarch64")]
+        EditVmStateSubCommand::RemoveRegs {
+            regs,
+            vmstate_path,
+            output_path,
+        } => edit(&vmstate_path, &output_path, |state| {
+            remove_regs(state, &regs)
+        })?,
+    }
+    Ok(())
+}
+
+fn edit(
+    vmstate_path: &PathBuf,
+    output_path: &PathBuf,
+    f: impl Fn(MicrovmState) -> Result<MicrovmState, EditVmStateError>,
+) -> Result<(), EditVmStateError> {
+    let (microvm_state, version) = open_vmstate(vmstate_path)?;
+    let microvm_state = f(microvm_state)?;
+    save_vmstate(microvm_state, output_path, version)?;
+    Ok(())
+}
+
+#[cfg(target_arch = "aarch64")]
+fn remove_regs(
+    mut state: MicrovmState,
+    remove_regs: &[u64],
+) -> Result<MicrovmState, EditVmStateError> {
+    for vcpu_state in state.vcpu_states.iter_mut() {
+        let mut new_regs = Aarch64RegisterVec::default();
+        for reg in vcpu_state
+            .regs
+            .iter()
+            .filter(|reg| !remove_regs.contains(&reg.id))
+        {
+            new_regs.push(reg);
+        }
+        vcpu_state.regs = new_regs;
```

Comment:
Maybe we can print a warning if some register's removal was requested, but the register isnt present?

Reply:
Added printing of which registers were removed and which regs were missing/not present.